internal/pkg/checks/pathRulesCheck.go

```diff
@@ -135,4 +241,102 @@ var pathRulesBarSubpathCheck = &Check{
 	},
 }
 
-// TODO: Implement more checks on edge cases like leading `/`, query params, and encoding
+var pathRulesPrefixAaaBbbcccCheck = &Check{
+	Name:        "path-rules-prefix-aaa-bbbccc",
+	Description: "Ingress with prefix path rule with a trailing slash should not match string prefix (/aaa/bbb/ does not match /aaa/bbbccc)",
+	Run: func(check *Check, config Config) (success bool, err error) {
```

Comment:
It feels like we can remove some of the boilerplate here if we specialize for the common case (but leave a generic Run function for advanced cases)
```
RunRequest: &Request{
  Path: xxx,
  Check: func(req ..., res ...) (*AssertionSet, error) {
    a := &AssertionSet{}
    ...
    return a, nil
  }
}
```

Reply:
This was captured in https://github.com/kubernetes-sigs/ingress-controller-conformance/issues/18